### DISCUSSION TITLE
num_samples in make_evaluation_predictions for DeepAr, how to fix the outcome prediction?

### DISCUSSION THREAD

**Discussion by Msaleh87** [awslabs/gluonts#2246] [Q&A] [answered]
I have an electric load data for about a year, I trained the model for a data set from Mar 2021 until Mar 2022. After I trained my deepAr model, whenever I change num_samples, I get a different prediction. My understanding is that num_samples is the average of different probabilistic paths (like the ave of multiple forecasts). Is there a way to get somehow fixed prediction? When I increase num_samples to 100 or more, the forecast somehow stays the same, however, I no longer have the fluctuations details of the profile.

Here is my code:
```
estimator = DeepAREstimator(freq = '5min',
                           context_length = 7*24*12,  #The previous data points I look at to predict. It's like n_forecast in Neural Prophet 
                           prediction_length = 1*24*12,
                           num_layers = 2,
                           num_cells = 64,
                           cell_type = 'lstm',
                           use_feat_dynamic_real =True,
                           trainer = Trainer(epochs= 50))

from gluonts.dataset.common import ListDataset
training_data = ListDataset([{"start": df_input.index[0],
                              "target": df_input.Load[:'2022-03-07 04:50:00'],
                              'feat_dynamic_real': [df_input.DofW[:'2022-03-07 04:50:00']]}],
                           freq = "5min")

predictor = estimator.train(training_data = training_data)

test_data = ListDataset(
    [
        {"start": df_input.index[0], "target": df_input.Load[:'2022-03-08 04:50:00'], 'feat_dynamic_real': [df_input.DofW[:'2022-03-08 04:50:00']]}, # one day
        {"start": df_input.index[0], "target": df_input.Load[:'2022-03-09 04:50:00'], 'feat_dynamic_real': [df_input.DofW[:'2022-03-09 04:50:00']]}   # two days
    
    ],

freq = "5min")

forecast_it, ts_it = make_evaluation_predictions(
    dataset = test_data,
    predictor = predictor,
    num_samples = 100 #
)
```

**Comment by lostella** [comment, marked as answer]
You should try setting the seed of MXNet’s random number generator, before doing predictions. See: https://mxnet.apache.org/versions/1.6/api/python/docs/api/mxnet/random/index.html#module-mxnet.random

**Reply by Msaleh87** [reply]
Thanks for getting back to me on this.
So I checked the link, but I don't get why this would be helpful. I understand that there is a random initialization when it comes to training the model, but how this would help? Also, How to include it in the code to train a DeepAr model? Thanks

**Reply by Msaleh87** [reply]
So I get the first figure when num_samples = 100
I get the second figure when I use reduced num_samples. Like 2 -->10, but every time I run the  code with reduced num_samples, let's say 2, I get a different profile, even though the num_samples is the same.

I understand that num_samples is like ensample, some sort of taking the average of different probabilistic paths, but how I can make the forecast profile somehow fixed with low num_samples, or is this even possible? Or there is something I am missing here? 
The way I imagined it is that u have P_1 path/forecast, P_2 path ... P_num_sample, then num_samples = (P1 + P2 + ... P_num_samples)/num_samples, so if I choose num_samples = 2 then I should get (P1+P2)/2

**Reply by Msaleh87** [reply]
![Figure 1 (8)](https://user-images.githubusercontent.com/42920113/187501763-c2cec6bd-d1eb-46eb-97a5-578faf2c8c28.png)

![Figure 19](https://user-images.githubusercontent.com/42920113/187503064-019eabf8-7bdc-47b9-bfd7-a9fc9a3d2ffd.png)
